File: tarefas-compartilhadas/src/main/java/br/edu/tarefascompartilhadas/model/Amizade.java
package br.edu.tarefascompartilhadas.model;

import br.edu.tarefascompartilhadas.enums.StatusAmizade;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.PrePersist;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Amizade {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "usuario_solicitante_id", nullable = false)
    private Usuario usuarioSolicitante;

    @ManyToOne
    @JoinColumn(name = "usuario_amigo_id", nullable = false)
    private Usuario usuarioAmigo;

    @Enumerated(EnumType.STRING)
    private StatusAmizade statusAmizade;

    private LocalDateTime dataSolicitacao;

    @PrePersist
    public void prePersist() {
        if (statusAmizade == null) {
            statusAmizade = StatusAmizade.PENDENTE;
        }
        if (dataSolicitacao == null) {
            dataSolicitacao = LocalDateTime.now();
        }
    }
}

File: app_tarefas_compartilhadas/lib/models/amizade.dart
import 'usuario.dart';

class Amizade {
  final int? id;
  final Usuario? usuarioSolicitante;
  final Usuario? usuarioAmigo;
  final String? statusAmizade;
  final String? dataSolicitacao;

  Amizade({
    this.id,
    this.usuarioSolicitante,
    this.usuarioAmigo,
    this.statusAmizade,
    this.dataSolicitacao,
  });

  factory Amizade.fromJson(Map<String, dynamic> json) {
    return Amizade(
      id: json['id'],
      usuarioSolicitante: json['usuarioSolicitante'] != null
          ? Usuario.fromJson(json['usuarioSolicitante'])
          : null,
      usuarioAmigo: json['usuarioAmigo'] != null
          ? Usuario.fromJson(json['usuarioAmigo'])
          : null,
      statusAmizade: json['statusAmizade'],
      dataSolicitacao: json['dataSolicitacao'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'usuarioSolicitante': usuarioSolicitante?.toJson(),
      'usuarioAmigo': usuarioAmigo?.toJson(),
      'statusAmizade': statusAmizade,
      'dataSolicitacao': dataSolicitacao,
    };
  }
}